frontend/apps/reader/modules/readerdictionary.lua

```diff
@@ -138,37 +134,55 @@ function ReaderDictionary:init()
         end
         logger.dbg("found", #available_ifos, "dictionaries")
 
+        if not G_reader_settings:readSetting("dicts_order") then
+            G_reader_settings:saveSetting("dicts_order", {})
+        end
+
+        self:sortAvailableIfos()
+
         if not G_reader_settings:readSetting("dicts_disabled") then
             -- Create an empty dict for this setting, so that we can
             -- access and update it directly through G_reader_settings
             -- and it will automatically be saved.
             G_reader_settings:saveSetting("dicts_disabled", {})
         end
     end
-    -- Prepare the -u options to give to sdcv if some dictionaries are disabled
+    -- Prepare the -u options to give to sdcv the dictionary order and if some are disabled
     self:updateSdcvDictNamesOptions()
     if not lookup_history then
         lookup_history = LuaData:open(DataStorage:getSettingsDir() .. "/lookup_history.lua", { name = "LookupHistory" })
     end
 end
 
+function ReaderDictionary:sortAvailableIfos()
+    local dicts_order = G_reader_settings:readSetting("dicts_order")
+
+    table.sort(available_ifos, function(lifo, rifo)
+        local lord = dicts_order[lifo.file]
+        local rord = dicts_order[rifo.file]
+
+        -- Both ifos without an explicit position -> lexical comparison
+        if lord == rord then
+            return util.strcoll(lifo.name, rifo.name)
+        end
+
+        -- Ifos without an explicit position come last.
+        return lord ~= nil and (rord == nil or lord < rord)
+    end)
+end
+
+
 function ReaderDictionary:updateSdcvDictNamesOptions()
-    self.enabled_dict_names = nil
+    self.enabled_dict_names = {}
 
     -- We cannot tell sdcv which dictionaries to ignore, but we
     -- can tell it which dictionaries to use, by using multiple
     -- -u <dictname> options.
-    -- (The order of the -u does not matter, and we can not use
-    -- them for ordering queries and results)
+    -- The order of the -u options controls the dictionary order
+    -- that sdcv uses to order its results.
```

Comment:
Really ?! Does that really work now?
I wrote that comment 2017-08-29.
We last updated sdcv around 2017-10 (to 0.5.2) and I added a few patches related to -u. Did I miss the fact that it would now follow the order of the -u ? :/

Btw, sdcv is now at 0.5.3.
(And re-btw, I still use our faster previous version 0.4 :/)

Reply:
According to the [changelog](https://github.com/Dushistov/sdcv/blob/e73388c726430ad724104b4a16d6c1c9821f35e9/NEWS#L17) and the [corresponding commit](https://github.com/Dushistov/sdcv/pull/11/commits/2d95bd0b12e5c977f1f4cb69ad3b5dc881a318ec) the `dictionary order by -u switches` feature was added in version 0.5.1.

Reply:
OK, it indeed works ! :)